src/query/functions/utils/heap.go

```diff
@@ -82,7 +99,7 @@ func NewFloatHeap(isMaxHeap bool, capacity int) FloatHeap {
 }
 
 // Push pushes a value and index pair to the heap
-func (fh FloatHeap) Push(value float64, index int) {
+func (fh *FloatHeap) Push(value float64, index int) {
```

Comment:
Do these need to be pointers?

Reply:
I think it's more clear this way because most of our functions mutates internal state. Also, generally it is advised to use either pointer receivers or value receivers on a type, not a mixture of both.

Reply:
None of these mutate the floatheap itself, just the internal data structure; these can all be reverted

Reply:
As I was describing before, I've done this for better clarity and intent. 
Also, I am not the only one thinking like this: https://github.com/golang/go/wiki/CodeReviewComments#receiver-type 

Here it states exactly the same situation we have here:
> If the receiver is a struct, array or slice and any of its elements is a pointer to something that might be mutating, prefer a pointer receiver, as it will make the intention more clear to the reader.

I can for sure revert it to be non pointer, but why this would better and what are the downsides of the current approach?